src/sections/shared/form/DatasetMetadataForm/useSubmitDataset.ts

```diff
@@ -56,11 +57,12 @@ export function useSubmitDataset(
     )
 
     if (mode === 'create') {
+      const DRAFT_PARAM = DatasetVersionNonNumericSearchParam.DRAFT
       createDataset(datasetRepository, formattedFormValues, collectionId)
         .then(({ persistentId }) => {
           setSubmitError(null)
           setSubmissionStatus(SubmissionStatus.SubmitComplete)
-          navigate(`${Route.DATASETS}?persistentId=${persistentId}`, {
+          navigate(`${Route.DATASETS}?persistentId=${persistentId}&version=${DRAFT_PARAM}`, {
```

Comment:
Just a suggestion, if DRAFT_PARAM is only going to be used in one place I think we don't need to add it to a constant variable, just using the enum directly in the url seems fine to me. e.g: 
```
navigate(
  `${Route.DATASETS}?persistentId=${persistentId}&version=${DatasetVersionNonNumericSearchParam.DRAFT}`,
  {
    state: { created: true },
  }
)
```